### PR TITLE
Remove unused client in rollout status

### PR DESCRIPTION
**Pull request by seans3**
* Removes unused client code within the rollout status functionality.

```release-note
NONE
```

### CODE REVIEW

**Comment by apelisse** [comment]
Good catch, clean change!

File: pkg/kubectl/rollout_status.go
Comment (soltysh) [inline pkg/kubectl/rollout_status.go]: While you're at it, please switch that `apps` to use `appsv1` here and in the next ones. That will remove yet another link with k8s internals.

**Review by soltysh — APPROVED** [review]
One nit, that I'd like to see addressed before this merges.

**Comment by soltysh** [comment]
I'm removing lgtm for now to get that nit fixed, feel free to self-lgtm once that is fixed.

File: pkg/kubectl/rollout_status.go
Comment (seans3) [reply pkg/kubectl/rollout_status.go]: Fixed that in a separate PR: https://github.com/kubernetes/kubernetes/pull/68641

**Pull request merged**
